keps/sig-cluster-lifecycle/kubeadm/2067-rename-master-label-taint/README.md

```diff
@@ -0,0 +1,470 @@
+and make progress.
+-->
+
+- Hard rename the kubeadm "master" label and taint keys to "control-plane"
+within a single release.
+- Leave users in the dark about the change.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (optional)
+
+#### Administrator
+
+As an administrator of kubeadm-created clusters, I manage additional critical
+workloads on Nodes labeled and tainted with "node-role.kubernetes.io/master".
+This change affects me and I would like to transition my infrastructure within
+a grace period.
+
+#### Administrator (single node)
+
+As an administrator, I deploy single-control-plane Node k8s clusters.
+This means that I have to remove taints with the key "node-role.kubernetes.io/master".
+This change affects me and I would like to transition my infrastructure within
+a grace period.
+
+#### Developer
+
+As an application or addon developer, I create deployments that tolerate the
+"node-role.kubernetes.io/master"-keyed taint or label-select the
+"node-role.kubernetes.io/master"-labeled Nodes. This change affects me and I would
+like to adapt my manifests within a grace period.
+
+#### Developer (higher up the stack)
+
+As a developer of a tool that sits on top of kubeadm in the stack, I have to adapt
+my source code to handle the renaming of the "master" label / taint. Managed by my tool,
+a version of kubeadm that no longer includes the "master" label / taint would require
+special handling.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+None
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+#### Risk 1
+
+Users not having enough visibility about the change, which results in
+breaking their setup.
+
+_Mitigation_
+
+Make sure the change is announced on all possible channels:
+- Include "action-required" release notes in all appropriate stages
+of the change.
+- Notify #kubeadm and #sig-cluster-lifecycle channels on k8s slack.
+- Notify the SIG Cluster Lifecycle and kubernetes-dev mailing lists.
+- Ask Twitter / Reddit users to post about the change.
+
+## Design Details
+
+The process will be broken into multiple stages:
+- Primary - 1.20
```

Comment:
@fabriziopandini 
i saw no objections for the primary stage in the draft.
this means we can merge this KEP as "implementable" for 1.20 and amend in later cycles.

if you agree i can update it as "implementable" in this PR.

Reply:
I'm +1 to kick off this process in 1.20